### PR TITLE
[IMP] l10n_co_edi: adapted to the flows of account_edi

### PR DESCRIPTION
**Pull request by bfr-o**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account_edi/models/account_edi_document.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ class AccountEdiDocument(models.Model):
     attachment_id = fields.Many2one('ir.attachment', help='The file generated by edi_format_id when the invoice is posted (and this document is processed).')
     state = fields.Selection([('to_send', 'To Send'), ('sent', 'Sent'), ('to_cancel', 'To Cancel'), ('cancelled', 'Cancelled')])
     error = fields.Html()
+    error_level = fields.Selection(selection=[('info', 'Info'), ('warning', 'Warning'), ('error', 'Error')])
```
Comment (smetl) [inline addons/account_edi/models/account_edi_document.py:21]:
```
    message = fields.Html()
    message_level = fields.Selection(selection=[('info', 'Info'), ('warning', 'Warning'), ('error', 'Error')])
``` 
?

File: addons/account_edi/models/account_move.py [hunk 2]
```diff
@@ -41,19 +44,32 @@ def _compute_edi_state(self):
...
+            levels = self.env['account.edi.document']._fields['error_level'].selection
+            strings = []
+            for level, string in levels:
+                err = len(errors.filtered(lambda d: d.error_level == level))
+                if err:
+                    strings.append(_('%s %s', err, string))
+                    if level == 'error':
+                        move.edi_blocked = err
+
+            move.edi_error_message = ', '.join(strings)
```
Comment (smetl) [inline addons/account_edi/models/account_move.py:62]: It's difficult to figure out how it looks like. I was thinking about something:
```
<div>
  <span>CFDI EDI: </span><span class="info-alert">error alert panic run away!</span>
  ...
</div>
```

File: addons/account_edi/tests/common.py [hunk 3]
```diff
@@ -60,10 +60,15 @@ def create_invoice_from_file(self, module_name, subfolder, filename):
         journal_id = self.company_data['default_journal_sale']
         invoice = journal_id.with_context(default_move_type='in_invoice')._create_invoice_from_single_attachment(attachment)
 
+    def get_generated_file(self, invoice):
+        return invoice.edi_document_ids.filtered(lambda d: d.edi_format_id == self.edi_format).attachment_id
```
Comment (smetl) [inline addons/account_edi/tests/common.py:64]: Could be usefull in `account.move`: `_get_edi_document(self, edi_format)` and/or `_get_edi_attachment(self, edi_format)`

File: addons/account_edi/models/account_move.py [hunk 4]
```diff
@@ -242,7 +260,11 @@ def message_post(self, **kwargs):
     ####################################################
 
     def action_process_edi_web_services(self):
-        self.edi_document_ids.filtered(lambda d: d.state in ('to_send', 'to_cancel'))._process_documents_web_services()
+        self.edi_document_ids.filtered(lambda d: d.state in ('to_send', 'to_cancel') and d.error != 'error')._process_documents_web_services()
+
+    def action_retry_documents_error(self):
```
Comment (smetl) [inline addons/account_edi/models/account_move.py:265]: `action_retry_edi_documents_error`

File: addons/account_edi/models/account_edi_document.py (hunk 1 above)
Comment (bfr-o) [reply addons/account_edi/models/account_edi_document.py:21]: I would prefer to keep 'error' as these are shown in a special tree view via the banner. 'Message' would confuse the developer making them think that they should return a message when like 'EDI successfully submitted to government', and the banner would show (polluting the view) whereas they should instead post it themselves on the chatter to keep track.

File: addons/account_edi/models/account_move.py (hunk 2 above)
Comment (bfr-o) [reply addons/account_edi/models/account_move.py:62]: This is actually done in the tree view. This is only the message appearing in the header ('Electronic Invoicing: 1 error, 2 warnings') => I renamed the field so it's more clear that it's the header.

**Pull request closed without merging**